src/agent/ipc/ipc_workers/tcp_worker_impl.cc

```diff
@@ -189,6 +189,7 @@ SpTcpWorker::Inject(SpChannel* c,
   sp_debug("INJECT CMD -- %s", cmd_exe.c_str());
 
   // Execute the command
+  unsetenv("LD_PRELOAD");
```

Comment:
This is a blunt hammer. Are we guaranteed this code will not be run in a user's environment (i.e., only in some child process)?

Reply:
Could you elaborate on scenarios where this can have problems.
Note that SPI itself is a library, so one way of using SPI is through assigning the environment variable LD_PRELOAD to load the agent library upon starting the process. The reason I included this statement is because we do not want instrumentation to follow into injector. Because injector will be spawned as a child process and LD_PRELOAD will follow into injector.

Reply:
I'm not familiar with how processes are spawned and handled in SPI, but whatever binary contains the `tcp_worker_impl` code should not modify the callee's environment. As long as that's not happening in this case, then we're all good.

Reply:
@tylergu Instead of unsetting `LD_PRELOAD`, you should first record what value `LD_PRELOAD` has and then remove injector library from `LD_PRELOAD`. The application code may require `LD_PRELOAD` some other libraries to run.

Reply:
I see the problem. Thank you.